### PR TITLE
Properly sort versions (closes #207 & #213)

### LINKED ISSUE #207: `takeout enable mysql` getting the lowest docker image version instead of latest
When leaving docker tag empty (default `latest`) or explicitly typing "latest", it picks the lowest version from Docker Hub instead...
Selecting the version explicitly works, however.

Note this is downloading `mysql:5.6.51` instead of `mysql:latest` (version `8.0.23` as of today)

```
➤ takeout enable mysql

 Which host port would you like mysql to use? [3306]:
 > 

 Which tag (version) of mysql would you like to use? [latest]:
 > latest

 What is the Docker volume name? [mysql_data]:
 > 

 What will the root password be? (null by default) []:
 > 

Downloading docker image...

 OUT  5.6.51: Pulling from library/mysql
 OUT  8aff230071c9: Pulling fs layer
 OUT  134fc34c9927: Pulling fs layer
 OUT  27dfb473d52e: Pulling fs layer
 OUT  702c333a167e: Pulling fs layer
 OUT  699bc078b452: Pulling fs layer
 OUT  01dd862365bd: Pulling fs layer
 OUT  7dbfc4425b5a: Pulling fs layer
 OUT  038f3579b073: Pulling fs layer
 OUT  d03e96fd56ed: Pulling fs layer
 OUT  01dd862365bd: Waiting
 OUT  d4e40c94cc3e: Pulling fs layer
 OUT  552099682f91: Pulling fs layer
 OUT  7dbfc4425b5a: Waiting
 OUT  d03e96fd56ed: Waiting
 OUT  038f3579b073: Waiting
 OUT  552099682f91: Waiting
 OUT  d4e40c94cc3e: Waiting
 OUT  702c333a167e: Waiting
 OUT  699bc078b452: Waiting
 OUT  134fc34c9927: Verifying Checksum
 OUT  134fc34c9927: Download complete
 OUT  27dfb473d52e: Verifying Checksum
 OUT  27dfb473d52e: Download complete
 OUT  8aff230071c9: Verifying Checksum
 OUT  8aff230071c9: Download complete
 OUT  8aff230071c9: Pull complete
 OUT  134fc34c9927: Pull complete
 OUT  27dfb473d52e: Pull complete
 OUT  699bc078b452: Verifying Checksum
 OUT  699bc078b452: Download complete
 OUT  702c333a167e: Download complete
 OUT  702c333a167e: Pull complete
 OUT  699bc078b452: Pull complete
 OUT  01dd862365bd: Verifying Checksum
 OUT  01dd862365bd: Download complete
 OUT  01dd862365bd: Pull complete
 OUT  7dbfc4425b5a: Download complete
 OUT  7dbfc4425b5a: Pull complete
 OUT  038f3579b073: Download complete
 OUT  d4e40c94cc3e: Verifying Checksum
 OUT  d4e40c94cc3e: Download complete
 OUT  038f3579b073: Pull complete
 OUT  d03e96fd56ed: Verifying Checksum
 OUT  d03e96fd56ed: Download complete
 OUT  d03e96fd56ed: Pull complete
 OUT  d4e40c94cc3e: Pull complete
 OUT  552099682f91: Verifying Checksum
 OUT  552099682f91: Download complete
 OUT  552099682f91: Pull complete
 OUT  Digest: sha256:a1868bd2d1c8c32a46baa7d06f38c3de76e949c8f811a5bc42bac6738520f3a7
 OUT  Status: Downloaded newer image for mysql:5.6.51
 OUT  docker.io/library/mysql:5.6.51
Enabling mysql...

 OUT  3c1dba027fa20b0eb6d39c7d88866ec8bd7692262fa14227c96ab2aba14ec8b6
111

### LINKED ISSUE #213: Install latest version of Postgres?
I enabled postgres and chose `latest`, but it pulled down version 10, whereas docker has up to version 13. It would be great if the 'latest' was at least what is installed by Forge, which I believe is version 12.

How can I help implement this? I took a browse through the code but didn't immediately see where the versions were retrieved.

Thanks

### PR DESCRIPTION
**Pull request by edalzell**
Closes #207
Closes #213

Added a failing test as well.

### CODE REVIEW

File: app/Shell/DockerTags.php [hunk 1]
```diff
@@ -44,7 +44,12 @@ public function getTags(): Collection
     {
         $response = json_decode($this->getTagsResponse()->getContents(), true);
 
-        return collect($response['results'])->map->name->filter();
+        return collect($response['results'])
```
Comment (mattstauffer) [inline app/Shell/DockerTags.php:47]: I'm a little nervous about sorting it ourselves; we may miss some Docker-specific versioning constraints that we're not aware of yet.

Was it not possible to pass the query parameter in that I mentioned?

Comment (edalzell) [reply app/Shell/DockerTags.php:47]: That did not sort properly, unfortunately. If you look here: https://registry.hub.docker.com/v2/repositories/library/postgres/tags?order=name you will see version `13.2` is not at the top nor the bottom.

Comment (edalzell) [reply app/Shell/DockerTags.php:47]: I also tried `ordering=name` with the same result. If you look in #207 you'll see the list that comes back isn't in the proper order.

Comment (mattstauffer) [reply app/Shell/DockerTags.php:47]: Ahh, bummer. OK--let's push this into `main` and then some folks can test it out before we tag it.

Thanks!

File: app/Shell/DockerTags.php [hunk 2]
```diff
@@ -44,7 +44,12 @@ public function getTags(): Collection
...
+            ->map(function ($item) {
```
Comment (mattstauffer) [inline app/Shell/DockerTags.php:48]: OK, so in this one you're turning it into something like this, right?

```
[
    ['name' => '8.4.0'],
    ['name' => '8.3.1'],
]
```

Could pluck the name first and then sort? I don't have the `sortDesc` code in front of me, so I don't know if it takes the sort type like sortbydesc does... but something like this?

```php
return collect($response['results'])
    ->pluck('name')
    ->sortDesc(SORT_NATURAL)
    ->filter();
```

Comment (edalzell) [reply app/Shell/DockerTags.php]: done

**Comment by mattstauffer** [comment]
Thanks @edalzell!

**Comment by mattstauffer** [comment]
@edalzell Sorry, but either something has changed or I didn't understand this when I was merging. The test you added is failing, as you mentioned, but... I guess what I assumed you meant was "I added a test that failed but now succeeds."

But.. you definitely left a test that just fails. What was your intention with that?

**Comment by edalzell** [comment]
> @edalzell Sorry, but either something has changed or I didn't understand this when I was merging. The test you added is failing, as you mentioned, but... I guess what I assumed you meant was "I added a test that failed but now succeeds."
> 
> But.. you definitely left a test that just fails. What was your intention with that?

Ya that's what I meant, it passed *after* my update. It breaks now? I can take a look.

**Comment by edalzell** [comment]
Ya if you look at my test and `it_lists_10_newest_available_tags_for_service` they are practically the same. The first one (mysql) passes and mine, `postgres` fails.

Perhaps the postgres stuff changed?

**Comment by edalzell** [comment]
Looks like `latest` is NOT in the most recent ten tags, as seen [here](https://hub.docker.com/_/postgres/?tab=tags&page=1&ordering=last_updated)

**Comment by edalzell** [comment]
Ok I have a fix @mattstauffer, let's discuss?

The issue is that the `latest` is not in the first 10. It turns out you can set `?page_size=20` and get more results (from [here](https://stackoverflow.com/a/56878013)).

Once you do that, the tags come back sorted properly again, and the tests pass (once you change the `10` test to `20`, of course).
 
Sound good?

**Comment by edalzell** [comment]
> Once you do that, the tags come back sorted properly again, and the tests pass (once you change the `10` test to `20`, of course).

Note that ALL tests continue to pass after getting 20 tags back, instead of ten.

**Pull request merged**